File: android/sdk/src/main/java/app/organicmaps/sdk/routing/LaneInfo.java
package app.organicmaps.sdk.routing;

import androidx.annotation.NonNull;

public final class LaneInfo
{
  public final LaneWay[] mLaneWays;
  public final LaneWay mActiveLaneWay;
  // How many adjacent identical physical lanes this entry stands for (see the C++
  // routing::turns::lanes::CollapseLanes); render one icon with a count badge when > 1.
  public final int mSimilarLanesCount;

  public LaneInfo(@NonNull LaneWay[] laneWays, LaneWay activeLane)
  {
    this(laneWays, activeLane, 1);
  }

  public LaneInfo(@NonNull LaneWay[] laneWays, LaneWay activeLane, int similarLanesCount)
  {
    mLaneWays = laneWays;
    mActiveLaneWay = activeLane;
    mSimilarLanesCount = similarLanesCount;
  }

  @NonNull
  @Override
  public String toString()
  {
    StringBuilder sb = new StringBuilder();
    sb.append("LaneInfo{activeLaneWay=").append(mActiveLaneWay.toString()).append(", laneWays=[");
    for (LaneWay i : mLaneWays)
      sb.append(" ").append(i);
    sb.append("], similarLanesCount=").append(mSimilarLanesCount).append("}");
    return sb.toString();
  }
}

File: libs/routing/lanes/lane_info.cpp
#include "lane_info.hpp"

#include <sstream>

namespace routing::turns::lanes
{
std::string DebugPrint(LaneInfo const & laneInfo)
{
  std::stringstream out;
  out << "LaneInfo{" << DebugPrint(laneInfo.laneWays) << ", recommendedWay: " << DebugPrint(laneInfo.recommendedWay)
      << "}";
  return out.str();
}

std::string DebugPrint(LanesInfo const & lanesInfo)
{
  std::stringstream out;
  out << "LanesInfo[";
  for (auto const & laneInfo : lanesInfo)
    out << DebugPrint(laneInfo) << ", ";
  out << "]";
  return out.str();
}
}  // namespace routing::turns::lanes
